### DISCUSSION TITLE
Ipad Connection

### DISCUSSION THREAD

**Discussion by sdstronic** [lathoub/Arduino-AppleMIDI-Library#106] [Q&A] [answered]
Hi, and first "great job" on Midi Works, many thanks
I would expérimente rtpmidi with esp32 on a "inteligent bridge" i work on.
all seems fine, but not soo, and i think i need smale  help :)
My conf: 
ESP32 running Democode and sending note ON/OFF every Second and monitoring receive note on Serial Monitor
My PC: runnink Protokol for monitoring, VMPK to send Data to "My PC"(rtp session) and rtpMidi for Conf
My Ipad 4 with MIDI Wrench
i have one session "my PC" with ESP and Ipad Connected.
When i Play on VMPK, note go to esp and ipad -> Cool
When i play on MIDIWrench , note go to "my PC" but not on ESP
and  Note ON/OFF play from ESP32 go to PC but not ipad.
how to set direct ESP32 -> Ipad, nothing found in ipad like rtpmidi ?

**Comment by sdstronic** [comment, marked as answer]
i reply to my own message, that could help :)
just had 
IPAddress remote(ip de l'ipad);
AppleMIDI.sendInvite(remote, DEFAULT_CONTROL_PORT); // port is 5004 by default
and it work fine !!

**Comment by svanimisetti** [comment]
@sdstronic - You can try [MIDI network](https://apps.apple.com/us/app/midi-network/id1102428729) by Oscar Bouwmans. It will create a MIDI connection with ESP32/ESP8266 directly using IP address and port. It is available for both iPhone and iPad - and I use it frequently.
